src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/receivers.kt

```diff
@@ -0,0 +1,168 @@
+        val directoryFile = directoryPath.toFile()
+        val directoryName = directoryFile.name
+
+        directoryExistingCheck(directoryFile)
+        if (!directoryFile.canWrite()) {
+            throw IOException("tasksDir directory(name is $directoryName) isn't writable")
+        }
+
+        this.tasksDir = directoryPath
+    }
+
+    fun unzipCompileTask(taskId: String, rootFileName: String, zipBytes: ByteString): File {
+        val destination = this.tasksDir.resolve(taskId).resolve("files")
+        val zipStream = ZipInputStream(ByteArrayInputStream(zipBytes.toByteArray()))
+        var entry: ZipEntry? = zipStream.nextEntry
+
+        while (entry != null) {
+            val filename = entry.name
+            val newFile = destination.resolve(filename).toFile()
+
+            if (!newFile.parentFile.exists() && !newFile.parentFile.mkdirs()) {
+                val dirName = newFile.parentFile.name
+                throw IOException("In task(id = $taskId): unable to create $dirName directory while unzipping")
+            }
+
+            FileOutputStream(newFile).use {
+                ByteStreams.copy(zipStream, it)
+            }
+
+            entry = zipStream.nextEntry
+        }
+
+        val rootFile = destination.resolve(rootFileName).toFile()
+        if (!rootFile.exists()) {
+            throw IOException("In task(id = $taskId): path to root file doesn't exists")
+        }
+
+        return rootFile
+    }
+
+    private fun compileProject(request: CompilerFleet.CompilerFleetRequest): File {
+        val rootFileFullPath = request.rootFileName
+        val zippedProject = request.zipBytes
+        val engine = request.engine
+
+        val rootFile = unzipCompileTask(request.taskId, rootFileFullPath, zippedProject)
+        return dockerProcessor.compileRmdToPdf(rootFile)
+    }
+
+    override fun processMessage(message: PubsubMessage): Boolean {
+        val request = CompilerFleet.CompilerFleetRequest.parseFrom(message.data)
+        val taskId = request.taskId
+        var isPublished = true
+
+        val compiledBytes: ByteString
+        val outputFileName: String
+
+        if (request.compiler == CompilerFleet.Compiler.MOCK) {
+            compiledBytes = ByteString.copyFrom(MOCK_PDF_BYTES)
+            outputFileName = MOCK_FILE_NAME
+        } else {
+            val compiledFile = compileProject(request)
+            outputFileName = compiledFile.name
+            compiledBytes = ByteString.copyFrom(FileUtils.readFileToByteArray(compiledFile))
+        }
+
+        val onPublishFailureCallback = {
+            LOGGER.info("Publish $taskId failed with code ${StatusCode.FAILURE}")
+            isPublished = false
+        }
+
+        val data = getResultData(taskId, compiledBytes, outputFileName, StatusCode.SUCCESS.ordinal)
+        resultPublisher.publish(data, onPublishFailureCallback)
+
+        return isPublished
+    }
+}
+
+class ConvertingTaskReceiver(
```

Comment:
Why is it called "converting"? As far as I can see, it does not convert anything.

Reply:
Added rough representation of converting